Data/ByteString/Short/Internal.hs

```diff
@@ -615,12 +615,33 @@ useAsCStringLen bs action =
 -- @since 0.11.3.0
 isValidUtf8 :: ShortByteString -> Bool
 isValidUtf8 sbs@(SBS ba#) = accursedUnutterablePerformIO $ do
-  i <- cIsValidUtf8 ba# (fromIntegral (length sbs))
+  let n = length sbs
+#if MIN_VERSION_base(4,10,0)
```

Comment:
There is some similar CPP for `isByteArrayPinned#` in `fromShort`. It would be cleaner to common this up into a compatibility shim/wrapper function that just conservatively returns `False` when the primop is unavailable. That should also resolve the unused import warnings.

Reply:
I have now made a `isPinned` compatibility function.